packages/manager/apps/pci-cold-archive/src/api/hooks/useUsers.ts

```diff
@@ -0,0 +1,291 @@
+    queryFn: () => getAllUsers(projectId),
+  });
+
+export const useUsers = (
+  projectId: string,
+  filterWithCredentials?: boolean,
+) => {
+  const { data: users, isPending } = useAllUsers(projectId);
+
+  const allUsersQueries = useQueries({
+    queries: (users || [])?.map((user) => ({
+      queryKey: [...getUsersCacheKey(projectId), user.id, 's3Credentials'],
+      queryFn: () => getS3Credentials(projectId, `${user.id}`),
+      enabled: !isPending,
+      select: (s3Credentials: TS3Credentials[]) => s3Credentials[0],
+    })),
+    combine: (results) => ({
+      isPending: results.some((result) => result.isPending),
+      isLoading: results.some((result) => result.isLoading),
+      error: results.find((result) => result.error),
+      data: users?.reduce<TUser[]>((all, user) => {
+        const s3Credentials = results.find(
+          (result) => result.data?.userId === user.openstackId,
+        )?.data;
+
+        if (filterWithCredentials && !s3Credentials) return all;
+
+        all.push({
+          ...user,
+          access: s3Credentials?.access,
+          s3Credentials,
+          search: `${user.username} ${user.description} ${user.creationDate} ${
+            s3Credentials ? s3Credentials?.access : ''
+          }`.trimEnd(),
+        });
+
+        return all;
+      }, []),
+    }),
+  });
+
+  const allUsers = allUsersQueries.data;
+  const validUsers = allUsers?.filter((user) => user.status === 'ok');
+  const validUsersWithCredentials = validUsers?.filter((user) => !!user.access);
+  const validUsersWithoutCredentials = validUsers?.filter(
+    (user) => !user.access,
+  );
+
+  return {
+    ...allUsersQueries,
+    validUsers,
+    validUsersWithCredentials,
+    validUsersWithoutCredentials,
+  };
+};
+
+export const usePaginatedUsers = (
+  projectId: string,
+  pagination: PaginationState,
+  sorting: ColumnSort,
+  filters: Filter[],
+) => {
+  const { validUsersWithCredentials, error, isLoading, isPending } = useUsers(
+    projectId,
+    true,
+  );
+
+  return useMemo(
+    () => ({
+      isLoading,
+      isPending,
+      paginatedUsers: paginateResults<TUser>(
+        sortResults<TUser>(
+          applyFilters<TUser>(validUsersWithCredentials || [], filters),
+          sorting,
+        ),
+        pagination,
+      ),
+      refetch: () => invalidateGetUsersCache(projectId),
+      error,
+    }),
+    [
+      validUsersWithCredentials,
+      error,
+      isLoading,
+      isPending,
+      pagination,
+      sorting,
+      filters,
+    ],
+  );
+};
+
+type DeleteUserProps = {
+  projectId: string;
+  userId: number;
+  access: string;
+  onError: (cause: Error) => void;
+  onSuccess: () => void;
+};
+
+export const useUser = (projectId: string, userId: number) =>
+  useQuery({
+    queryKey: getUserCacheKey(projectId, userId),
+    queryFn: async () => {
+      const user = await getUser(projectId, userId);
+      const s3Credentials = await getS3Credentials(projectId, `${user.id}`);
```

Comment:
This should be replaced with a useQueries and 2 different queries since they can run in parallel (user.id is the same as userId)